rust/src/res.rs

```diff
@@ -0,0 +1,195 @@
+use ::std::error::Error;
+use ::std::fmt;
+use ::std::fs::File;
+use ::std::io::Seek;
+use ::std::path::Path;
+
+use ::serde::{Deserialize, Serialize};
+use ::serde_json::{json, Map, Value};
+
+use crate::slf::{start_of_entries, SlfHeader, SlfEntryState};
+
+
+// A pack of game resources.
+#[derive(Debug, Default, Deserialize, Serialize)]
+pub struct ResourcePack {
+    // A friendly name of the resource pack for display purposes.
+    pub name: String,
+
+    // The properties of the resource.
+    //
+    // # Properties
+    //
+    //  * with_archive_slf (bool, default = false) - the contents of SLF archives are included
+    pub properties: Map<String,Value>,
```

Comment:
Could we make the string an enum here, so we don't have arbitrary properties?

Reply:
see https://github.com/ja2-stracciatella/ja2-stracciatella/pull/820#issuecomment-486643636